bridges/snowbridge/pallets/inbound-queue-v2/src/lib.rs

```diff
@@ -244,63 +195,35 @@ pub mod pallet {
 			// Verify that the message was submitted from the known Gateway contract
 			ensure!(T::GatewayAddress::get() == envelope.gateway, Error::<T>::InvalidGateway);
 
-			// Retrieve the registered channel for this message
-			let channel =
-				T::ChannelLookup::lookup(envelope.channel_id).ok_or(Error::<T>::InvalidChannel)?;
-
-			// Verify message nonce
-			<Nonce<T>>::try_mutate(envelope.channel_id, |nonce| -> DispatchResult {
-				if *nonce == u64::MAX {
-					return Err(Error::<T>::MaxNonceReached.into())
-				}
-				if envelope.nonce != nonce.saturating_add(1) {
-					Err(Error::<T>::InvalidNonce.into())
-				} else {
-					*nonce = nonce.saturating_add(1);
-					Ok(())
-				}
-			})?;
-
-			// Reward relayer from the sovereign account of the destination parachain, only if funds
-			// are available
-			let sovereign_account = sibling_sovereign_account::<T>(channel.para_id);
-			let delivery_cost = Self::calculate_delivery_cost(message.encode().len() as u32);
-			let amount = T::Token::reducible_balance(
-				&sovereign_account,
-				Preservation::Preserve,
-				Fortitude::Polite,
-			)
-			.min(delivery_cost);
-			if !amount.is_zero() {
-				T::Token::transfer(&sovereign_account, &who, amount, Preservation::Preserve)?;
-			}
+			// Verify the message has not been processed
+			ensure!(!<Nonce<T>>::contains_key(envelope.nonce), Error::<T>::InvalidNonce);
 
-			// Decode payload into `VersionedMessage`
-			let message = VersionedMessage::decode_all(&mut envelope.payload.as_ref())
+			// Decode payload into `MessageV2`
+			let message = MessageV2::decode_all(&mut envelope.payload.as_ref())
```

Comment:
Lets make use of rust modules as namespaces, instead of appending `V2` to type names.

```suggestion
			let message = Message::decode_all(&mut envelope.payload.as_ref())
```

Reply:
It is actually just called Message, but aliased to MessageV2 here: https://github.com/claravanstaden/polkadot-sdk/pull/3/files#diff-bfe761b371a553619edf5be2dd792c45bd14f1a916e51b77f5b77b9395d92cdaR52 since I use `Message` from `snowbridge_core` too.